### PR TITLE
test: throw on deprecation

### PR DESCRIPTION
**Pull request by JustinBeckwith**

### CODE REVIEW

**Comment by ofrobots** [comment]
@DominicKramer the failure in tests is due to an unhandled promise rejection. This needs to be investigated.

**Comment by JustinBeckwith** [comment]
@ofrobots @googleapis/node-team can someone take a look?  Would love to get this merged :)

**Comment by ofrobots** [comment]
@DominicKramer what's the status here?

**Comment by fhinkel** [comment]
ping @DominicKramer

**Comment by JustinBeckwith** [comment]
*gentle ping*

**Comment by DominicKramer** [comment]
The design of the test require the use of unhandled rejections to verify that the library handles them correctly.  There were also test failures unrelated to this PR that I am investigating so that I can get this PR landed.

**Comment by DominicKramer** [comment]
The changes in PR #253 will fix the system tests and update the tests so that this PR can land.

**Comment by JustinBeckwith** [comment]
@DominicKramer Looks like the system tests are still failing here :/  https://source.cloud.google.com/results/invocations/fa56d7ab-1577-4496-b7dc-cd8de37c3048/targets/cloud-devrel%2Fclient-libraries%2Fnodejs%2Fgoogleapis%2Fnodejs-error-reporting%2Fpresubmit%2Fnode8%2Fsystem-test/log

**Comment by JustinBeckwith** [comment]
@DominicKramer 👋

File: test/unit/google-apis/auth-client.ts [hunk 1]
```diff
@@ -60,6 +61,7 @@ function verifyReportedMessage(
   // tslint:disable-next-line:no-unused-expression
   new RequestHandler(config2, logger);
   setImmediate(() => {
+    console.log('TOAST');
```
Comment (stephenplusplus) [inline test/unit/google-apis/auth-client.ts]: 🍞 🔥

**Pull request closed without merging**